covid_graphs/covid_graphs/country_graph.py

```diff
@@ -26,50 +26,48 @@ def __init__(
             self,
             data_dir: Path,
             country_predictions: List[CountryPrediction],
-            graph_type: GraphType = GraphType.Normal,
     ):
-        self.graph_type = graph_type
-
         # TODO(miskosz): We assume there is only one country.
         # This might change soon though if we want to have a country dropdown in one graph.
-        country_name = country_predictions[0].country
+        self.short_name = country_predictions[0].country
+        self.min_case_count = min(prediction.formula.min_case_count
+                                  for prediction in country_predictions)
 
         # TODO(lukas): Figure out better strategy for more predictions
         if len(country_predictions) >= 1:
             self.prediction_date = country_predictions[0].prediction_event.date.strftime("%Y-%m-%d")
 
-        # Due to plotly limitations, we can only have graphs with dates on the x-axis when we
-        # aren't using logs.
-        axis_type = XAxisType.Dated if graph_type == GraphType.Normal else XAxisType.Numbered
-
-        report = CountryReport(country_data_file=data_dir / f'{country_name}.data')
+        report = CountryReport(country_data_file=data_dir / f'{self.short_name}.data')
+        self.long_name = report.name
 
         first_idx, last_idx, self.curves = Curve.create_curves(
             [prediction.formula for prediction in country_predictions],
             report.cumulative_active,
             report.dates[0],
-            axis_type,
         )
-        self.name = report.name
-        self.min_case_count = min(prediction.formula.min_case_count
-                                  for prediction in country_predictions)
-
         self.cumulative_active = report.cumulative_active[first_idx:].copy()
         self.date_list = report.dates_str[first_idx:]
-        if axis_type == XAxisType.Dated:
-            self.t = self.date_list
-        else:
-            self.t = np.arange(len(self.cumulative_active)) + 1
+        self.t = np.arange(len(self.cumulative_active)) + 1
 
-    def create_country_figure(self):
-        colors = ['SteelBlue', 'Purple', 'Green'][:len(self.curves)]
+    def create_country_figure(self, graph_type: GraphType):
+
+        # Due to plotly limitations, we can only have graphs with dates on the x-axis when we
+        # aren't using logs.
+        axis_type = XAxisType.Dated if graph_type == GraphType.Normal else XAxisType.Numbered
 
+        def pick_xaxis_labels(object):
+            if axis_type == XAxisType.Dated:
+                return object.date_list
+            else:
+                return object.t
```

Comment:
This is a little bit funny but I guess it is ok.

Reply:
We're working around plotly's bugs - do you have suggestions for a better approach?